File: Task-3/Wonderful-Number.js
// Function to check if a string is a palindrome
function isPalindrome(str) {
  const reversed = str.split("").reverse().join("");
  return str === reversed;
}

// Function to check if a number is wonderful
function isWonderful(N) {
  if (N % 2 === 0) {
    return "NO";
  }

  // Convert N to binary and check if it's a palindrome
  const binaryRepresentation = N.toString(2);
  if (isPalindrome(binaryRepresentation)) {
    return "YES";
  } else {
    return "NO";
  }
}

// Example usage
const N1 = 3;
console.log(isWonderful(N1));

const N2 = 4;
console.log(isWonderful(N2));

File: Task-3/Wonderful Number.c
#include <stdio.h>
#include <stdbool.h>

bool isOdd(int num)
{
    return num % 2 != 0;
}

bool isPalindrome(int num)
{
    int reversed = 0;
    int original = num;

    while (num > 0)
    {
        reversed = (reversed << 1) | (num & 1);
        num = num >> 1;
    }

    return original == reversed;
}

void checkWonderful(int N)
{
    if (isOdd(N) && isPalindrome(N))
    {
        printf("YES\n");
    }
    else
    {
        printf("NO\n");
    }
}

int main()
{
    int N;
    scanf("%d", &N);

    checkWonderful(N);

    return 0;
}
